### PR TITLE
User Goals Updates

### PR DESCRIPTION
**Pull request by kwiedenhaupt**
Functionality of these changes on the profile/v2 page:
Screen Shot 2022-12-07 at 8 30 39 AM

These changes: 
1. retain the styling for the highlighted goals, etc. that is on the main goals page
2. expand the view of the goal so that it extends to the entire screen
3. Takes in a prop that filters to only current goals and slices to only three goals
4. retains the same functionality for the goals page
5. eliminates the need for ProfileGoalsSection that is used on the main profile page and profile/v2 page -- BUT the profile page would need to be updated like the profile/v2 page with the fetchUserGoals query to populate the data field.

Functionality of these changes on the goals page:
Screen Shot 2022-12-07 at 8 31 59 AM

Eventually the goal would be to implement this GoalsSectionComponent.tsx and get rid of ProfileGoalsSection.tsx.

### CODE REVIEW

**Review by iamnotreddy — CHANGES_REQUESTED** [review]
- let's rewrite the returnGoalStyle function so that it takes in a goal of type UserGoalsData and move it outside the component definition... this way we can move it outside the map function (idk why I put it there in the first place) and we can use it across the repo if we so choose

- we can just call differenceInCalendarDays from date-fns directly, think it's overkill to define a separate function

- I don't think we actually need abridgedUserGoals. How about we load userGoals on the profile similar to [this](https://github.com/skillify-ca/skillify-web/blob/2e0042c298d7f711aee9d52a948699fbc1473227/pages/goals/index.tsx#L42-L65) but only load the "Current Section" and do the slicing here (so we only get the three most recent goals). This way our component is just rendering whatever data we supply it, not much logic going on inside

- let's keep using redux for this part since we already have it set up for goals

I don't think we break anything with this approach. Down the line, let's think about where / when we're making our Hasura calls because I'm seeing some inefficiency... currently if you go to the profile page, we query all your goals, then if you navigate to the goal page we query all your goals again. Don't need to solve for that in this PR but I think we should eventually

**Comment by iamnotreddy** [comment]
One other note - I think we can handle showing fewer fields on the mobile view with tailwind [visibility classes](https://tailwindcss.com/docs/visibility) + [mobile breakpoints](https://tailwindcss.com/docs/screens)

**Comment by kwiedenhaupt** [comment]
All changes requested were made/addressed. This is currently the mobile vs. PC view.
Screen Shot 2022-12-08 at 10 41 31 AM
Screen Shot 2022-12-08 at 10 41 44 AM

File: components/coding/GoalsSectionComponent.tsx [hunk 1]
```diff
@@ -1,18 +1,32 @@
...
 import { UserGoalsData } from "../../graphql/fetchUserGoals";
-
 export type GoalsSectionProps = {
-  userGoals: UserGoalsData[];
-  sectionName: string;
+  sectionName?: string;
+  userGoals?: UserGoalsData[];
+};
+
+export const returnGoalStyle = (goal) => {
```
Comment (iamnotreddy) [inline components/coding/GoalsSectionComponent.tsx:11]: let's define the type for the function input so Typescript knows what's available on goal.. 

i.e const returnGoalStyle = (goal: UserGoalsData)...

File: pages/profile/v2/[slug].tsx [hunk 2]
```diff
@@ -1,12 +1,15 @@
 import { ApolloClient, InMemoryCache } from "@apollo/client";
+import GoalsSectionComponent from "../../../components/coding/GoalsSectionComponent";
```
Comment (iamnotreddy) [inline pages/profile/v2/[slug].tsx:2]: not sure why your linter didn't pick up these blank spaces

File: components/coding/profileV2/BadgesDisplayedComponent.tsx [hunk 3]
```diff
@@ -1,3 +1,4 @@
+
 import React from "react";
 
 import { User } from "./ProfileHeaderComponent";
```
Comment (iamnotreddy) [inline components/coding/profileV2/BadgesDisplayedComponent.tsx:4]: You moved this to fetchUserProfile so need to update the import path

Is your linter broken? I feel like it should alert you to changed imports and remove the random blank rows across your code

File: components/coding/GoalsSectionComponent.tsx [hunk 4]
```diff
@@ -27,44 +41,26 @@ export default function GoalsSection({
...
+              goal
             )}`}
           >
             <p>{index + 1}.</p>
             <p className="col-span-2 md:col-span-4">{goal.goalName}</p>
             <p className="hidden md:block md:col-span-2">
               {format(new Date(goal.createdAt), "MM/dd/yyyy")}
             </p>
-            <p className="md:col-span-2">
+            <p className="hidden md:block col-span-1 md:col-span-2">
```
Comment (iamnotreddy) [inline components/coding/GoalsSectionComponent.tsx:55]: nice!

File: components/coding/profileV2/ProfileHeaderComponent.tsx [hunk 5]
```diff
@@ -16,6 +17,7 @@ import BadgesDisplayedComponent from "./BadgesDisplayedComponent";
 import JoinedDateComponent from "./JoinedDateComponent";
 
 export type UserProfileSectionProps = {
+
```
Comment (iamnotreddy) [inline components/coding/profileV2/ProfileHeaderComponent.tsx:20]: not sure why your linter didn't pick up these blank spaces

**Review by iamnotreddy — APPROVED** [review]
a few changes but otherwise looks good! approved for now but please fix these before merging

nice work

File: components/coding/GoalsSectionComponent.tsx (hunk 1 above)
Comment (kwiedenhaupt) [reply components/coding/GoalsSectionComponent.tsx]: thank you! did that.

**Pull request merged**